src/tun.c

```diff
@@ -687,6 +764,90 @@ tun_setip(const char *ip, const char *other_ip, int netbits)
 		if_name, ip, inet_ntoa(net));
 	return system(cmdline);
 #endif
+
+}
+
+int
+tun_setip6(char *display_ip6, const char *display_other_ip6, int netbits6)
+{
+	int v6_r;
+        struct in6_addr ip6;
+        char v6_cmdline[512];
+        if (inet_pton(AF_INET6, display_ip6, &ip6) < 1){
+            warnx("Error in IPv6 address");
+        }
+
+#ifdef WINDOWS32
+        /*
+	DWORD status;
+	DWORD ipdata[3];
+	struct in_addr addr;
+	DWORD len;
+        */
+#endif
+
+	if (netbits6 > 0) {
+
+	   fprintf(stderr, "Setting IPv6 of %s to %s\n", if_name, display_ip6);
+
+#if defined LINUX
+                snprintf(v6_cmdline, sizeof(v6_cmdline),
+                        IFCONFIGPATH "ifconfig %s inet6 add %s/%d",
+                        if_name,
+                        display_ip6, netbits6);
+#else
+                snprintf(v6_cmdline, sizeof(v6_cmdline),
+                        IFCONFIGPATH "ifconfig %s inet6 %s/%d",
+                        if_name,
+                        display_ip6, netbits6);
+#endif
+
+	        v6_r = system(v6_cmdline);
+
+                if (v6_r != 0) {
+                    return v6_r;
+                } else {
+		    return 0;
+		}
+        }
+
+#ifdef WINDOWS32 /* WINDOWS32 */
+
+	/* Set device as connected */
+	fprintf(stderr, "Enabling interface '%s'\n", if_name);
```

Comment:
Is this really needed? Already done when setting IPv4

Looks like a straight copy from `tun_setip`?